app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,80 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'tweets/create_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post'/tweets' do
+    if params[:content] == ""
+      redirect to '/tweets/new'
+    else
+      user = User.find_by_id(session[:user_id])
+      @tweet = Tweet.create(:content=> params[:content], :user_id=> user.id)
+
+      redirect("/tweets/#{@tweet.id}")
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
+       erb :'tweets/edit_tweet'
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content] == ""
+      redirect to "/tweets/#{params[:id]}/edit"
+    else
+      @tweet = Tweet.find_by_id(params[:id])
+      @tweet.content = params[:content]
```

Comment:
a better way to do lines 58-60 would be to use the active record update method and if successful do seperate redirects. 

```ruby
if @tweet.update(content: params[:content])
  redirect to "/tweets/#{@tweet.id}"
else 
  redirect to "/tweets/#{@tweet.id}/edit"
end
```